document/core/exec/modules.rst

```diff
@@ -59,15 +59,17 @@ The following auxiliary typing rules specify this typing relation relative to a
 :math:`\EVMEM~a`
 ................
 
-* The store entry :math:`S.\SMEMS[a]` must be a :ref:`memory instance <syntax-meminst>` :math:`\{\MIDATA~b^{n\cdot64\,\F{Ki}}, \MIMAX~m^?, \MISHARE~s\}`, for some :math:`n`.
+.. todo:: refactor semantics to handle shared instances
 
-* Then :math:`\EVMEM~a` is valid with :ref:`external type <syntax-externtype>` :math:`\ETMEM~(\{\LMIN~n, \LMAX~m^?\}~s)`.
+* The store entry :math:`S.\SMEMS[a]` must be a :ref:`memory instance <syntax-meminst>` :math:`\{\MITYPE~\{\LMIN~n, \LMAX~m^?\}~s, \MIDATA~b^{k\cdot64\,\F{Ki}}\}`.
+
+* Then :math:`\EVMEM~a` is valid with :ref:`external type <syntax-externtype>` :math:`\ETMEM~\{\LMIN~k, \LMAX~m^?\}~s`.
 
 .. math::
    \frac{
-     S.\SMEMS[a] = \{ \MIDATA~b^{n\cdot64\,\F{Ki}}, \MIMAX~m^?, \MISHARE~s\}
+     S.\SMEMS[a] = \{ \MITYPE~\{\LMIN~n, \LMAX~m^?\}~s, \MIDATA~b^{k\cdot64\,\F{Ki}} \}
```

Comment:
What's the reasoning for this change? It's a little strange that the memory instance now has a "minimum" component which is never used, and it makes the definition of external typing more surprising. The change is also not currently mirrored in the table instance, although it could be.

It does make the soundness proof somewhat neater (give the related change to memory instance validity). Is that why?

Reply:
I just realised the change to the syntax of the memory instance was made before and I missed it - I should have asked this question them